dhcpv4/nclient4/lease_test.go

```diff
@@ -236,13 +236,21 @@ func (sll *testServerLeaseList) runTest(t *testing.T) {
 		sll.lastTestSvrErrLock.RLock()
 		keepgoing := chkerr(err, sll.lastTestSvrErr, l.ShouldFail, t)
 		sll.lastTestSvrErrLock.RUnlock()
+
 		if keepgoing {
-			err = clnt.Release(lease)
-			//this sleep is to make sure release is handled by server
-			time.Sleep(time.Second)
+			lease, err = clnt.Renew(context.Background(), lease.Offer)
 			sll.lastTestSvrErrLock.RLock()
-			chkerr(err, sll.lastTestSvrErr, l.ShouldFail, t)
+			keepgoing = chkerr(err, sll.lastTestSvrErr, l.ShouldFail, t)
 			sll.lastTestSvrErrLock.RUnlock()
+
+			if keepgoing {
```

Comment:
Do we need this as a nested if? Can it not go outside of the previous if?

Reply:
I think what was intended here is to stop progressing if an error occurs, but make sure to always unlock the mutex, thus the nested `if`s. I can however refactor it to use `continue`s instead to avoid the nesting.